src/main/java/sparksoniq/spark/udf/OrderClauseCreateColumnsUDF.java

```diff
@@ -120,19 +122,46 @@ public Row call(WrappedArray<byte[]> wrappedParameters) {
 
                     // extract type information for the sorting column
                     String typeName = _allColumnTypes.get(expressionIndex);
-
-                    if (typeName.equals("bool")) {
-                        _results.add(nextItem.getBooleanValue());
-                    } else if (typeName.equals("string")) {
-                        _results.add(nextItem.getStringValue());
-                    } else if (typeName.equals("integer")) {
-                        _results.add(nextItem.castToIntegerValue());
-                    } else if (typeName.equals("double")) {
-                        _results.add(nextItem.castToDoubleValue());
-                    } else if (typeName.equals("decimal")) {
-                        _results.add(nextItem.castToDecimalValue());
-                    } else {
-                        throw new SparksoniqRuntimeException("Unexpected ordering type found while creating columns.");
+                    try {
+                        switch (typeName) {
+                            case "bool":
```

Comment:
is "bool" instead of "boolean" an overlook from the past? (all other types correspond to JSONiq types).

Reply:
Yes, it looks like an overlook from the past. Fixed